main/src/cgeo/geocaching/connector/su/SuApi.java

```diff
@@ -0,0 +1,187 @@
+                "minlat", String.valueOf(viewport.getLatitudeMin()),
+                "maxlat", String.valueOf(viewport.getLatitudeMax()),
+                "minlng", String.valueOf(viewport.getLongitudeMin()),
+                "maxlng", String.valueOf(viewport.getLongitudeMax()));
+
+        final JSONResult result = SuApi.getRequest(connector, SuApiEndpoint.CACHE_LIST, params);
+        return SuParser.parseCaches(result.data);
+    }
+
+    private static String getSuLogType(final LogType logType) {
+        switch (logType) {
+            case FOUND_IT:
+                return "found";
+            case DIDNT_FIND_IT:
+                return "notFound";
+            case NOTE:
+            default:
+                return "comment";
+        }
+    }
+
+    @NonNull
+    public static LogResult postLog(@NonNull final Geocache cache, @NonNull final LogType logType, @NonNull final Calendar date, @NonNull final String log) {
+        final IConnector connector = ConnectorFactory.getConnector(cache.getGeocode());
+        if (!(connector instanceof SuConnector)) {
+            return new LogResult(StatusCode.LOG_POST_ERROR, "");
+        }
+        final SuConnector gcsuConnector = (SuConnector) connector;
+
+        final Parameters params = new Parameters();
+        params.add("cacheID", cache.getCacheId());
+        params.add("type", getSuLogType(logType));
+        params.add("text", log);
+
+        final ObjectNode data = postRequest(gcsuConnector, SuApiEndpoint.NOTE, params).data;
+
+        if (data == null) {
+            return new LogResult(StatusCode.LOG_POST_ERROR, "");
+        }
+        if (data.get("status").get("code").toString().contains("ERROR")) {
+            return new LogResult(StatusCode.LOG_POST_ERROR, "");
+        }
+
+        return new LogResult(StatusCode.NO_ERROR, data.get("data").get("noteID").asText());
+    }
+
+    @NonNull
+    private static JSONResult getRequest(@NonNull final SuConnector connector, @NonNull final SuApiEndpoint endpoint, @NonNull final Parameters params) {
+        return request(connector, endpoint, "GET", params);
+    }
+
+    @NonNull
+    private static JSONResult postRequest(@NonNull final SuConnector connector, @NonNull final SuApiEndpoint endpoint, @NonNull final Parameters params) {
+        return request(connector, endpoint, "POST", params);
+    }
+
+    @NonNull
+    private static JSONResult request(@NonNull final SuConnector connector, @NonNull final SuApiEndpoint endpoint, @NonNull final String method, @NonNull final Parameters params) {
+        final String host = connector.getHost();
+        if (StringUtils.isBlank(host)) {
+            return new JSONResult("unknown connector host");
+        }
+
+        final SuLogin suLogin = SuLogin.getInstance();
+
+        //TODO: Try to re-login if needed ?
+//        if (suLogin.oAtoken.isEmpty() && !suLogin.getActualUserName().isEmpty()) {
+//            suLogin.login(true);
+//        }
```

Comment:
Commented code?

Reply:
When I was testing\debugging it, it was fine, but this weekend I went "to the field" and found out that after some time some error happens so c:geo unable to send the log, for example. After manual re-login it was fine. So this piece here is just a reminder that maybe it will be required. Or, if I found out the source of this API behavior and see the root cause, we can remove it.

Reply:
done